packages/core/src/browser/shell/application-shell.ts

```diff
@@ -307,15 +307,15 @@ export class ApplicationShell extends Widget {
         this.options = {
```

Comment:
IMO we could resolve the options directly in the constructor. There is no need to this in postConstruct because we don't rely on any injected services. This would also remove the need to expose the partial options as protected property.

Reply:
Thank you for the having a look! Good point, I updated the PR with this simplification.